include/openthread/border_routing.h

```diff
@@ -209,15 +220,40 @@ void otBorderRoutingClearRouteInfoOptionPreference(otInstance *aInstance);
  * Thread network if there isn't already an OMR prefix. This prefix can be reached
  * from the local Wi-Fi or Ethernet network.
  *
+ * Note: The local generated OMR prefix is returned. When DHCPv6 PD is enabled, the
+ * border router may publish the prefix from DHCPv6 PD.
+ *
  * @param[in]   aInstance  A pointer to an OpenThread instance.
  * @param[out]  aPrefix    A pointer to where the prefix will be output to.
  *
  * @retval  OT_ERROR_INVALID_STATE  The Border Routing Manager is not initialized yet.
  * @retval  OT_ERROR_NONE           Successfully retrieved the OMR prefix.
  *
+ * @sa otBorderRoutingGetPdOmrPrefix
+ *
  */
 otError otBorderRoutingGetOmrPrefix(otInstance *aInstance, otIp6Prefix *aPrefix);
 
+/**
+ * Gets the DHCPv6 Prefix Delegation (PD) provided off-mesh-routable (OMR) prefix.
+ *
+ * Only mPrefix, mValidLifetime and mPreferredLifetime fields are used in the returned prefix info.
+ *
+ * `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE` must be enabled.
+ *
+ * @param[in]   aInstance    A pointer to an OpenThread instance.
+ * @param[out]  aPrefixInfo  A pointer to where the prefix info will be output to.
+ *
+ * @retval  OT_ERROR_NONE           Successfully retrieved the OMR prefix.
+ * @retval  OT_ERROR_INVALID_STATE  The Border Routing Manager is not initialized yet.
+ * @retval  OT_ERROR_NOT_FOUND      There are no valid PD prefix on this BR.
+ *
+ * @sa otBorderRoutingGetOmrPrefix
+ * @sa otPlatBorderRoutingProcessIcmp6Ra
+ *
+ */
+otError otBorderRoutingGetPdOmrPrefix(otInstance *aInstance, otBorderRoutingPrefixTableEntry *aPrefixInfo);
```

Comment:
In which cases is this API useful? Why would the client cares whether the OMR prefix is a PD prefix? If they want to know if their OMR prefix has global scope, it can already be got by checking the scope of return value of `otBorderRoutingGetOmrPrefix`

Reply:
> Also, as described in implementation, receiving a ULA prefix as PD prefix is possible as the upper router might be providing NAT66 and NAT64 functions so we cannot determine the kind of the prefix by checking the return value of otBorderRoutingGetOmrPrefix.

But why does it matter of being a PD prefix to the client if they can't tell the difference?

Reply:
This is useful for testing and debugging.

For example, currently, we have `otBorderRoutingGetOmrPrefix` and `otBorderRoutingGetFavoredOmrPrefix`, for most use cases, users can just use `otBorderRoutingGetFavoredOmrPrefix` since they don't have to care about which router actually published this prefix as the packets can be sent to every BRs in the thread network.